python/ray/tests/test_scheduling_2.py

```diff
@@ -304,6 +306,51 @@ def task2():
         assert set(ray.get(locations)) == worker_node_ids
 
 
+def test_demand_report_when_scale_up(shutdown_only):
+    # https://github.com/ray-project/ray/issues/22122
+    from ray.cluster_utils import AutoscalingCluster
+
+    cluster = AutoscalingCluster(
+        head_resources={"CPU": 0},
+        worker_node_types={
+            "cpu_node": {
+                "resources": {
+                    "CPU": 1,
+                    "object_store_memory": 1024 * 1024 * 1024,
+                },
+                "node_config": {},
+                "max_workers": 10,
+            },
+        },
+    )
+
+    cluster.start()
+
+    info = ray.init("auto")
+
+    @ray.remote
+    def foo():
+        import time
+
+        time.sleep(999)
+
+    tasks = [foo.remote() for _ in range(10000)]  # noqa: F841
+    global_state_accessor = make_global_state_accessor(info)
+
+    def get_backlog_size():
+        message = global_state_accessor.get_all_resource_usage()
+        if message is None:
+            return 0
+
+        resource_usage = gcs_utils.ResourceUsageBatchData.FromString(message)
+        aggregate_resource_load = resource_usage.resource_load_by_shape.resource_demands
+        if len(aggregate_resource_load) == 1:
```

Comment:
can you assert that the shape is right here too? like it should have `{"CPU": 1}`.